### PR TITLE
Set the date as today only if it is not provided

### LINKED ISSUE #306: Sample collection date is always overridden as the particular day
Whenever status is made `SENT_TO_COLLECTON_CENTRE` . Mark it as the day only if the date is not provided.

See
```
        if validated_data.get("status") == PatientSample.SAMPLE_TEST_FLOW_MAP["SENT_TO_COLLECTON_CENTRE"]:
            validated_data["date_of_sample"] = make_aware(datetime.datetime.now())
```

in `PatientSamplePatchSerializer`'s `update` method.

### PR DESCRIPTION
**Pull request by agzuniverse**
Fixes #306

### CODE REVIEW

File: care/facility/api/serializers/patient_sample.py [hunk 1]
```diff
@@ -73,11 +73,14 @@ def update(self, instance, validated_data):
         if choice == "COMPLETED" and not validated_data.get("result"):
             raise ValidationError({"result": [f"is required as the test is complete"]})
 
-        if validated_data.get("status") == PatientSample.SAMPLE_TEST_FLOW_MAP["SENT_TO_COLLECTON_CENTRE"]:
+        if (
+            validated_data.get("status") == PatientSample.SAMPLE_TEST_FLOW_MAP["SENT_TO_COLLECTON_CENTRE"]
+            and validated_data.get("date_of_sample") is None
```
Comment (vigneshhari) [inline care/facility/api/serializers/patient_sample.py:78]: Codacy Issue found: [Wrong hanging indentation before block (add 4 spaces).](https://app.codacy.com/gh/coronasafe/care/pullRequest?prid=5344889)

File: care/facility/api/serializers/patient_sample.py [hunk 2]
```diff
@@ -73,11 +73,14 @@ def update(self, instance, validated_data):
         if choice == "COMPLETED" and not validated_data.get("result"):
             raise ValidationError({"result": [f"is required as the test is complete"]})
 
-        if validated_data.get("status") == PatientSample.SAMPLE_TEST_FLOW_MAP["SENT_TO_COLLECTON_CENTRE"]:
+        if (
+            validated_data.get("status") == PatientSample.SAMPLE_TEST_FLOW_MAP["SENT_TO_COLLECTON_CENTRE"]
```
Comment (vigneshhari) [inline care/facility/api/serializers/patient_sample.py:77]: Codacy Issue found: [Wrong hanging indentation before block (add 4 spaces).](https://app.codacy.com/gh/coronasafe/care/pullRequest?prid=5344889)

**Comment by anroopak** [comment]
@agzuniverse Can you write/update test cases for this?

File: care/facility/api/serializers/patient_sample.py (hunk 2 above)
Comment (agzuniverse) [reply care/facility/api/serializers/patient_sample.py:77]: This conflicts with the autoformatting applied by Black...

**Comment by agzuniverse** [comment]
@anroopak It doesn't look like there are any tests written for patient sample. You can merge this PR since it resolves the issue. I'll look into writing tests soon.

**Pull request closed without merging**